zeebe/atomix/cluster/src/test/java/io/atomix/raft/ControllableRaftContexts.java

```diff
@@ -396,7 +402,9 @@ public void restart(final MemberId memberId) {
   // of one node at a time.
   public void restartWithDataLoss(final MemberId memberId) {
     LOG.info("Shutting down member {}", memberId.id());
-    raftServers.get(memberId).close();
+    try (final var ignored = MDC.putCloseable("actor-scheduler", memberId.toString())) {
+      raftServers.get(memberId).close();
+    }
```

Comment:
:thought_balloon: Makes me think that the `close()` implementation should just schedule itself on the proper thread context.

Reply:
Originally I tried but didn't worked: the reason was that I logged "Raft context closed" after closing the ThreadContext. Logging before that makes it work